commit 5f37c04c7bf7b79cb0e3815a944e7613a0b54e7a (refs/changes/35/1635/3)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-03-09T18:06:55+00:00 (2 years, 7 months ago)
    
    TOOLS-1716 sdcadm fails with "no such image in local IMGAPI"

diff --git a/lib/steps/agent-services.js b/lib/steps/agent-services.js
index afd2df0..08ef937 100644
--- a/lib/steps/agent-services.js
+++ b/lib/steps/agent-services.js
@@ -91,35 +91,6 @@ function agentServicesEnsureCreated(arg, cb) {
         agentServices[n].metadata[logLevelKey] = 'info';
     });
 
-
-    // The first time we add agent services to SAPI we'll use the HN image
-    // version to create the service, assuming that's the version installed
-    // everywhere across the whole SDC setup
-    function getAgentImages(callback) {
-        vasync.forEachPipeline({
-            func: function (agent, next) {
-                var name = agent.name;
-                var imgUUIDPath = util.format(
-                    '/opt/smartdc/agents/lib/node_modules/%s/image_uuid',
-                    name);
-                fs.readFile(imgUUIDPath, {
-                    encoding: 'utf8'
-                }, function (err, data) {
-                    if (err) {
-                        log.error({err: err}, 'Error reading agent image uuid');
-                        next();
-                        return;
-                    }
-                    agentServices[name].params.image_uuid = data.trim();
-                    next();
-                });
-            },
-            inputs: agentNames.map(function (agent) {
-                return agentServices[agent];
-            })
-        }, callback);
-    }
-
     var newAgentServices = [];
     var updateAgentServices = [];
 
@@ -180,10 +151,6 @@ function agentServicesEnsureCreated(arg, cb) {
             }, next);
         },
 
-        function getAgentImgVersions(_, next) {
-            getAgentImages(next);
-        },
-
         function saveChangesToHistory(_, next) {
             var changes = [];
             newAgentServices.forEach(function (s) {
